### PR TITLE
Dependency updates

### PR DESCRIPTION
**Pull request by gerryster**
This updates the following dependencies:

1. Ruby to 3.0! This required no code changes other than the Ruby version in the gemspec.
2. The pre-compiled date-holidays Node module.
3. Ruby build CircleCI orb.
3. Ruby 2.5.7 to 2.5.8.
4. Node for development purposes the current LTS version.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -24,7 +24,7 @@ jobs:
               - ruby-orbs/bundle-install:
                   with_gemfile_lock: false
                   gemspec_name: date_holidays-reader
-                  cache_key_prefix: ruby-bundle-v1.0.1
+                  cache_key_prefix: ruby-version-<< parameters.ruby-version >>-bundle-v1.0.0
```
Comment (gerryster) [inline .circleci/config.yml:27]: Note that I parameterized this key with the Ruby version. I encountered a race condition where the Ruby 2.7 build failed as it ran after the 2.6 build and used its Gemfile.lock which specified Bundler version one. Ruby 2.7 ships with only Bundler v2 so it blew up.

File: Guardfile [hunk 2]
```diff
@@ -17,7 +17,7 @@
 #
 # and, you'll have to watch "config/Guardfile" instead of "Guardfile"
 
-# Note: The cmd option is now required due to the increasing number of ways
+# NOTE: The cmd option is now required due to the increasing number of ways
```
Comment (gerryster) [inline Guardfile:20]: New Rubocop rule.

File: date_holidays-reader.gemspec [hunk 3]
```diff
@@ -45,7 +45,9 @@ Gem::Specification.new do |spec|
   spec.add_development_dependency 'rake', '~> 13.0'
   spec.add_development_dependency 'rspec', '~> 3.0'
   spec.add_development_dependency 'rspec_junit_formatter'
-  spec.add_development_dependency 'rubocop', '~> 1.1'
+  spec.add_development_dependency 'rubocop', '~> 1'
+  spec.add_development_dependency 'rubocop-rake'
```
Comment (gerryster) [inline date_holidays-reader.gemspec:49]: Rubocop suggested these.

File: date_holidays-reader.gemspec [hunk 4]
```diff
@@ -32,7 +32,7 @@ Gem::Specification.new do |spec|
   spec.executables   = spec.files.grep(%r{^exe/}) { |f| File.basename(f) }
   spec.require_paths = ['lib']
 
-  spec.required_ruby_version = '~> 2.5'
+  spec.required_ruby_version = '>= 2.5'
```
Comment (gerryster) [inline date_holidays-reader.gemspec:35]: I thought about adding a '< 4.0' clause here. However, I don't want to be too restrictive. I have no way of knowing at this point that it /won't/ work with Ruby 4.0.

**Review by ddewar — APPROVED** [review]
Looks good to me.

**Pull request merged**